06-building-a-program-start-to-finish/writing-tests/output/spec/InvertedIndexSpec.js

```diff
@@ -0,0 +1,277 @@
+          text: `an unusual alliance of man, elf, dwarf, wizard
+     and hobbit seek to destroy a powerful ring.`,
+        },
+      ];
+      expect(InvertedIndex.lowerDocumentText(lowerDocumentFile))
+      .toEqual(result);
+    });
+  });
+
+  describe('Tests for the Search methods', () => {
+    const documentMock = { 'doc.json': [{ alice: [0],
+      alliance: [1],
+      an: [1],
+      man: [1],
+      hole: [0],
+      a: [0],
+      falls: [0],
+      rabbit: [0],
+      unusual: [1] }, ['Alice in Wonderland',
+        'The Lord of the Rings: The Fellowship of the Ring.']] };
+
+    it(`should return an object of words
+     that appear in a single indexed file`, () => {
+      const result = { alice: [0],
+        alliance: [1],
+        an: [1],
+        man: [1],
+        hole: [0],
+        a: [0],
+        falls: [0],
+        rabbit: [0],
+        unusual: [1] };
+      expect(InvertedIndex.searchIndex('doc.json', 'alice,an,hole a man',
+         documentMock)).toEqual(result);
+    });
+
+    it(`the buildSearchResult should build
+     and return correct search result`, () => {
+      const result = { alice: [0],
+        alliance: [1],
+        an: [1],
+        man: [1],
+        hole: [0],
+        a: [0],
+        falls: [0],
+        rabbit: [0],
+        unusual: [1] };
+      expect(InvertedIndex.buildSearchResult('doc.json', 'alice,an,hole a man',
+         documentMock)).toEqual(result);
+    });
+
+    it(`returns an object of file names against
+    their search results for a search of all files`, () => {
+      const allFilesResult = InvertedIndex.searchIndex('All Files',
+      'alice,an,hole,a - man', allFilesMock);
+      const result = { 'doc.json': {
+        alice: [0],
+        alliance: [1],
+        an: [1],
+        man: [1],
+        hole: [0],
+        a: [0],
+        falls: [0],
+        rabbit: [0],
+        unusual: [1] },
+
+        'test.json': { alice: [0],
+          alliance: [1, 2],
+          an: [1, 2],
+          and: [0, 1, 2],
+          man: [1, 2],
+          hole: [0],
+          a: [0, 1, 2],
+          dwarf: [1, 2],
+          falls: [0],
+          imagination: [0],
+          rabbit: [0],
+          unusual: [1, 2],
+          wizard: [1, 2] } };
+      expect(allFilesResult).toEqual(result);
+    });
+
+    it('should take search string in any case and return result', () => {
+      const result = {
+        alice: [0],
+        alliance: [1],
+        an: [1],
+        man: [1],
+        hole: [0],
+        a: [0],
+        falls: [0],
+        rabbit: [0],
+        unusual: [1] };
+      expect(InvertedIndex.searchIndex('doc.json', 'ALICE,AN,HOLE A MAN',
+         documentMock)).toEqual(result);
+    });
+
+    it('should return words containing substring in search query', () => {
+      const result = { alliance: [1],
+        an: [1],
+        man: [1] };
+      expect(InvertedIndex.searchIndex('doc.json', 'an',
+       documentMock)).toEqual(result);
+    });
+  });
+});
```

Comment:
Please ensure you do not have variable names that are have the numeral suffix or abbreviated.